index.html

```diff
@@ -251,19 +265,108 @@ <h2>Direct</h2>
                <code>@type</code> including the semantic term <code>Directory</code>.
             </p>
         </section>
+        <section id="introduction-well-known" class="normative">
+            <h2>Well-Known URIs</h2>
+            <p>
+                A Thing or Directory Service MAY use the Well-Known Uniform Resource Identifier [[RFC8615]]
+                to advertise its presence.  The Thing or Directory Service registers its own Thing Description
+                into the following path:
+                <code>/.well-known/wot-thing-description</code>.
+            <p>
+                When the HTTP GET access is made to the above path, the HTTP server MUST return
+                a Thing Description with the <code>content-type</code> set to <code>application/td+json</code>.
+            </p>
+            <p class="ednote" title="Registration of Well-known URI">
+                The service name in Well-Known URI (<code>wot-thing-description</code>) is tentative.
+                "Well-Known URIs" registry and contents of registration request is described in Section 3.1 of [[RFC8615]].
+            </p>
+
+        </section>
         <section id="introduction-dns-sd" class="normative">
-            <h2>DNS-SD</h2>
-            <p>To Do.
+            <h2>DNS-Based Service Discovery</h2>
+            <p>A Thing or Directory Service MAY use the DNS-Based Service Discovery (DNS-SD)[[RFC6763]].
+                This can be also be used to discover them on the same link by combining Multicast DNS (mDNS)[[RFC6762]].
+            </p>
+            <p>
+                In DNS-SD, format of the Service Instance Name is <code>Instance.Service.Domain</code>. 
+                The Service part is a pair of labels following the conventions of [[RFC2782]].
+                The first label has an underscore followed by the Service Name,
+                and the second label describes the protocol.
+            </p>
+            <p>
+                The Service Name to indicate the Thing MUST be <code>_thing._sub._wot</code>.
+                And the Service Name to indicate the Directory Service MUST be <code>_directory._sub._wot</code>.
```

Comment:
I think `_thing` is too abstract as a subtype for physical devices; directory is also a Thing. I suggest using `_device` and `_directory` subtypes instead.

Reply:
Also worth noting that the use of subtypes makes it possible to browse devices and directories separately or at the same time. However, when browsed at the same time, it won't be possible to tell them apart because subtype value won't appear in the results. This was discussed in https://github.com/w3c/wot-testing/issues/34. 

We can assume that all registrations are for devices and have an additional TXT record to distinguish directories.

Reply:
Really these should describe types of "exploration" services.   A directory is clearly one type, so we should have "directory" as one.    The other kind of exploration service though is when a Thing self-describes and hosts its own TD, so the type could be "self".   The other option is to leave off the subtype, but we should not imply that all things can self-describe.   It also needs to work with URL-only introductions... which should probably always point at a "self" endpoint.